### PR TITLE
feat: first version of custom FMGC, ND and flight plan manager

### LINKED ISSUE #939: [BUG] AP not following STAR or approach
**Mod Version**
master

**Describe the bug**
When I'm about to enter a STAR or approach (if I didn't chose a STAR) the AP stops following the flight path and just keeps flying straight ahead.

**To Reproduce**
1. Create an IFR flight plan including a STAR
2. Start the flight
3. Engage the autopilot in full managed mode

**Expected behavior**
The AP should navigate the aircraft along the flight path all the way to the last waypoint in the flight plan

**Actual behavior**
The AP stops navigating just before reaching the first waypoint in the STAR and keeps flying in a straight line.

**References**

**Additional context**
I'm using the latest release of the Navigraph beta for MSFS 2020, perhaps there's a problem with that rather than with the aircraft or this mod

Discord username (if different from GitHub): ZenIT_SWE

### LINKED ISSUE #1385: [REQUEST]F-PLN Discontinuity
**Is your feature request related to a problem? Please describe.**
The request relates to an inaccuracy in the f-pln page of the MCDU. When there are no waypoints entered between the dep/arr airports, there should be a flight-plan discontinuity. Instead, the f-pln page has no discontinuities and simply shows the deceleration waypoint.

**Describe the feature you'd like to see implemented**
Discontinuities wherever appropriate. They don’t seem to even exist in the default MCDU 

**References**
See message from IRL pilot Dokug in #mcdu channel on discord 

**Additional context**
N/A

Discord username (if different from GitHub):
NONAmr2433 [Z-5] #8777

### LINKED ISSUE #1388: [BUG] Cannot delete or edit waypoints in a STAR
**Mod Version**
0.4.0

**Describe the bug**
I insert an Simbrief created Flightplan in Sim and therefore also in A320 . 
I see the complete Plan with SID , TRANS and STAR. 
But all of the Waypoints are in Color blue not in green Color. 

**To Reproduce**
1. Create Flightplan in Simbrief
2.Load it in to FS2020
3.Start Flight

**Expected behavior**
All Waypoints in FMC Flightplan Page should be in Green Color and deleteable . 

**Actual behavior**
All Waypoints in FMC are in Blue Color. 
It is not possible to delete any of this Waypoints . 

**Why the Waypoints are blue and not Green ? Thanks** 

2020-10-16 07_11_18-Microsoft Flight Simulator - 1 9 5 0

**Here u can see this Function in P3D with FSLabs .** 

2020-10-16 09_28_54-Lockheed Martin® Prepar3D® v5 with WideServer_ waiting for clients

**All waypoints are deletable**

2020-10-16 09_31_26-Lockheed Martin® Prepar3D® v5 with WideServer_ waiting for clients

**References**

### LINKED ISSUE #2345: [BUG] Next closest Waypoint causes troubles
**Mod Version**

 latest Dev Master

**Describe the bug**

If next closest Waypoint is a RWY, the plane leaves the APP path and flies OVH RWY at APP altitude. Plane flies directly from DH609 to RW15. Same applies if next WP is farer away as last RWY WP during TO. If AP is engaged too early (in distance) Plane turns back at 45 bank to RWY, which is most likeliy reason for death roll (in non modified PID settings)

**To Reproduce**

1. Fly e.g. EDDH ILS RWY 15 via NOLGO1P!! I know its not the right APP, but is shows clearly the problem!
2.
3.

**Expected behavior**

Plane should follow APP or initial climb w/out shortcutting the path.

**Actual behavior**

**References**

**Additional context**

Was this working before/when did the issue start occurring?
Can't say, but long lasting

Is this a problem in the vanilla unmodded game?
n/a

Discord username (if different from GitHub):

### LINKED ISSUE #2350: [BUG] ? Following Flight Plan after arrival entered ? Possible user error?
**Mod Version**

Master

**Describe the bug**

When en route, following entry of arrival detals to the MCDU aircraft back tracks to previous waypoint.

**To Reproduce**

1. Carry out flight LGIR to LGAV
2. Approaching waypoint KEA, atc advises approach. 
3. Enter approach into MCDU and press INSERT

**Expected behavior**

Following press of INSERT aircraft should continue on to next way point.

**Actual behavior**

Aircraft commences return to KEA and then resumes plan (effectively 360 degree turn)

**References**

**Additional context**

Maybe an operator input problem but I have followed 320 Sim Pilots video tutorials.

Was this working before/when did the issue start occurring?

I feel this has been continuous since first stable release.

Is this a problem in the vanilla unmodded game?

Not known.

Discord username (if different from GitHub):

### LINKED ISSUE #2683: [BUG] MCDU arrival selection screen resets when new STAR transition selected
**Mod Version**

{
  "built": "2020-12-23T23:38:26+00:00",
  "ref": "",
  "sha": "",
  "actor": "DerL30N <leon.beeser@yahoo.de>",
  "event_name": "manual"
}

**Describe the bug**

If the user selects a different transition when loading a STAR into the MCDU, the MCDU sends the users back to the runway selection page. The user then has to reselect the runway, the STAR, and the VIA in-order to see the correct transition.

**To Reproduce**

1. Init any arrival airport that has a STAR with multiple transitions to choose from. (KCMH for example)
2. On the F-PLN page, select the arrival airport and select "arrival".
3. Select a runway with a STAR that has multiple transitions to choose from, then select that STAR. 
4. Select any "via"
5. Note the value that appears under "TRANS"
6. Select a transition other than the current value of "TRANS"

**Expected behavior**

The "TRANS" value should display the transition that the user selected. 

**Actual behavior**

MCDU sends the users back to the runway selection screen. The Approach and via still display, however the STAR and TRANS fields are now blank and need to be re-selected. Once re-selected the previously selected transition will be loaded properly. 

**References**

https://youtu.be/IQTxxVErWWo

**Additional context**

Was this working before/when did the issue start occurring?
Never was working

Is this a problem in the vanilla unmodded game?
Yes

Discord username (if different from GitHub): Mobius_One_Fox3

### LINKED ISSUE #2848: [BUG] Unable to insert STAR without transition
**Mod Version**
Development branch
```
{
  "built": "2020-12-31T18:05:31+00:00",
  "ref": "",
  "sha": "",
  "actor": "Brooke McKim <brookemckim@users.noreply.github.com>",
  "event_name": "manual"
}
```

**Describe the bug**
I generated the following flight plan using SimBrief but run into issues trying to enter it into the FMC. More specifically, it is very difficult to go direct from NELIE to VALRE, which is my first fix on the STAR>

```
ROUTING:

ROUTE ID: DEFRTE
KBOS/15R PATSS5 PATSS DCT NELIE DCT VALRE HAARP3 KLGA/04

--------------------------------------------------------------------
 
                             FLIGHT LOG                             
                             ----------                             

MOST CRITICAL MORA 03600 FEET AT NELIE///MXSHR 09 AT VALRE
--------------------------------------------------------------------
AWY                           FL   IMT   MN    WIND  OAT  EFOB  PBRN
POSITION    LAT      EET ETO MORA  ITT  TAS    COMP  TDV
IDENT       LONG    TTLT ATO DIS  RDIS   GS     SHR  TRP  AFOB  ABRN
FREQ
--------------------------------------------------------------------
                                   137                    12.6   0.4
LOGAN INTL  N4221.8      ...  22   121         P021     
KBOS       W07100.4 0000 ...       241  307               ....  ....

PATSS5                       087   091  .44 239/053  M01  12.2   0.8
FOXXX       N4218.9 0001 ...  21   076         P021  P01
FOXXX      W07054.2 0001 ...   5   236  307          396  ....  ....

PATSS5                       120   129  .56 253/056  M05  11.9   1.1
BRRRO       N4220.0 0002 ...  21   114         P056  P04
BRRRO      W07048.1 0003 ...   5   231  419          396  ....  ....

PATSS5                       137   202  .58 256/052  M06  11.6   1.4
JAITE       N4218.2 0001 ...  22   187         P040  P06
JAITE      W07042.8 0004 ...   4   227  415          396  ....  ....

PATSS5                       157   274  .61 258/058  M09  11.1   1.9
HEWMO       N4211.3 0002 ...  22   259         M023  P07
HEWMO      W07044.0 0006 ...   7   220  365          396  ....  ....

PATSS5                       160   274  .61 258/059  M10  11.0   2.0
T O C       N4210.9 0001 ...  23   259  389    M059  P07
           W07046.7 0007 ...   2   218  330       4  396  ....  ....

PATSS5                       160   278  .64 258/059  M10  10.8   2.2
KENWI       N4209.6 0001 ...  28   263  403    M059  P07
KENWI      W07056.4 0008 ...   7   211  344       4  396  ....  ....

PATSS5                       160   273  .64 258/060  M10  10.2   2.8
PATSS       N4205.3 0006 ...  33   258  403    M060  P07
PATSS      W07142.7 0014 ...  35   176  343       4  402  ....  ....

DCT                          160   254  .64 262/059  M11   9.3   3.7
NELIE       N4156.5 0008 ...  36   240  402    M059  P06
NELIE      W07241.3 0022 ...  45   131  343       3  401  ....  ....

DCT                          160   159  .64 251/063  M10   8.1   4.9
VALRE       N4126.1 0011 ...  32   145  403    M062  P07
VALRE      W07352.9 0033 ...  62    69  341       9  413  ....  ....

HAARP3                       160   159  .64 251/064  M10   8.0   5.0
BASYE       N4120.6 0001 ...  30   145  402    P013  P07
BASYE      W07347.9 0034 ...   7    62  415       9  412  ....  ....

HAARP3                       160   225  .64 251/064  M10   7.7   5.3
HAARP       N4107.0 0002 ...  27   212  402    P013  P07
HAARP      W07335.6 0036 ...  16    46  415       9  412  ....  ....

 
AWY                           FL   IMT   MN    WIND  OAT  EFOB  PBRN
POSITION    LAT      EET ETO MORA  ITT  TAS    COMP  TDV
IDENT       LONG    TTLT ATO DIS  RDIS   GS     SHR  TRP  AFOB  ABRN
FREQ
--------------------------------------------------------------------
HAARP3                       160   225  .64 251/064  M10   7.6   5.3
CRALY       N4104.6 0001 ...  25   212  403    M052  P07
CRALY      W07337.6 0037 ...   3    43  351       9  412  ....  ....

HAARP3                       160   225  .64 251/064  M10   7.5   5.5
T O D       N4059.5 0001 ...  23   212  403    M052  P07
           W07341.8 0038 ...   6    37  351       9  412  ....  ....

NEW YORK FIR/UIR                                                    
-KZNY       N4051.5 0005 ...
           W07348.4 0043 ...   9    28

HAARP3                       025   218  .39 173/023   06   7.4   5.6
LA GUARDIA  N4047.0 0003 ...  21   205         M018  M04
LGA        W07352.1 0046 ...   6    22  240          412  ....  ....
113.10

HAARP3                                                     6.9   6.1
LAGUARDIA   N4046.6 0008 ...                            
KLGA       W07352.4 0054 ...  22                          ....  ....
```

**To Reproduce**
1. Enter KBOS/KLGA on the INIT page of the FMC
2. Select RWY15R departure from KBOS using the PATSS5 SID
3. Enter NELIE as next waypoint from PATSS
4. Attempt to select HAARP3 arrival for RWY04 at KLGA without a transition. See attached video for additional details.

**Expected behavior**
If no transition is selected, it should select the primary fix on the STAR, in this case VALRE intersection.

**Actual behavior**
If I select the NONE option for TRANS, the entire STAR is removed. If I try again, select the star, and click INSERT without selecting a TRANS (so ----- is displayed), it inserts what was previously the default selection for TRANS, in this case ALB. The only way I can get the correct flight plan is to delete the waypoints between the incorrectly-added transition and VALRE.

**References**
1. See [video](https://onedrive.live.com/?authkey=%21ADwvIf%5F8%2DsT0XRI&cid=33EF0832ACC47842&id=33EF0832ACC47842%21144475&parId=33EF0832ACC47842%21144460&o=OneUp)
2. See [chart](https://i.imgur.com/Sjo9Ct1.png)

**Additional context**
I admit I am far from an expert on what the proper behavior should be here, but I'm reasonably confident that what's captured in my video is not correct.

### LINKED ISSUE #2893: [BUG] FPLN does not sequence properly
**Mod Version**
{
  "built": "2021-01-05T05:22:58+00:00",
  "ref": "",
  "sha": "",
  "actor": "donstim <70166617+donstim@users.noreply.github.com>",
  "event_name": "manual"
}
**Describe the bug**
Despite flying the route in NAV mode, the MCDU FPLN page does not sequence waypoints and delete them as the flight progresses. This is mirrored in the DIR TO page where it still gives you the option to DIR TO waypoints well behind you on the FPLN. 

**To Reproduce**
1. Load route IAW FCOM
2. Fly route in NAV
3. Once progressed down route, use arrow buttons and whole route still displayed

**Expected behavior**
For example flying between A B C D

Once you have passed A it should display A on 1L and B will display 2L. C and D 3L and 4L respectively. 
Once you have passed B it should display B on 1L and C will display 2L. D will display on 3L.

You should not be able to click back up the FPLN and find A. It should be deleted. 

**Actual behavior**
All FPLN waypoints remain in sequence

**References**
[https://app.photobucket.com/u/olibarke/p/6c0ac3db-a405-426a-9c53-960eda0e642f](url)

As can be seen I am at VUREP, yet I still have departure runway, SID, and route from in the past. Also as can be seen I have tried to DIR TO a sequenced waypoint and still has old route in FPLN page and showing on ND. 

**Additional context**
A330 type rated. Airbus experienced.

Is this a problem in the vanilla unmodded game? - Yes

Discord username (if different from GitHub): olibarke

### LINKED ISSUE #4175: Stuttering Dev Mod when set Flight Plan (Custom AP/ FADEC)
**Mod Version**

Unstable Master Branch Build (with custom FBW / Autopilot / FADEC)

**Describe the bug**

is that known that in the 3D Cockpit start stuttering when i set the flight plan? 

**To Reproduce**

1.Start Flight at the Gate 308 from EDDM cold and dark
2.set flight plan manually or with simbrief
3.taxi to runway and take off

**Expected behavior**

the stutetring starst when the flight plan is set

**Actual behavior**

stuttering when Taxi and in flight

**References**

**Additional context**

Was this working before/when did the issue start occurring?
no stuttering before Custom AP and FADEC integrates
Is this a problem in the vanilla unmodded game?
no

Discord username (if different from GitHub):

**BY REMOVING THIS LINE, I ATTEST THAT I HAVE FOLLOWED THE ISSUE TEMPLATE AND PROVIDED A TITLE, as well as searched for other possible duplicate issues. I acknowledge if this line is not removed and/or the issue template is not respected, my issue will be closed without warning**

### LINKED ISSUE #4682: F-PLAN deleted as MDCU change destination to origin
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-05-01T17:57:16+00:00",
  "ref": "master",
  "sha": "6e71264d7618dc1d98193a567e4036583d0b865b",
  "actor": "MisterChoker",
  "event_name": "manual"
}
```

### Describe the bug

When approaching KSFO from SERFR4 just before EDDYY MDCU deleted plan and set KLAX as destination.

### Expected behavior

MDCU sets EDDYY as next waypoint

### Steps to reproduce

Unknown. It happened once so far.

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### LINKED ISSUE #5037: ND: Current leg flight plan green line direction modified before passing a waypoint
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-06-06T15:12:39+00:00",
  "ref": "master",
  "sha": "ff9425ae19d607662e1e8adb6732dd0509aacb9f",
  "actor": "aguther",
  "event_name": "manual"
}
```

### Describe the bug

I follow a FP generated by LittleNavMap (See attached)

I am in cruise phase
I reach the waypoint LENSU from QPR (straight line)
On approching LENSU and when starting a turn for the next leg, the current leg is changed direction completely on the ND.
The angle between the 2 legs is highly changed.

Here are the snapshots:

snap103

snap104

snap105

snap106

snap107

It is systematic at this waypoint.I do not notice such behaviour in other waypoints of this FP.

The related flight plan:
[IFR Bretagne (LFRB) to St Jacques (LFRN).zip](https://github.com/flybywiresim/a32nx/files/6607529/IFR.Bretagne.LFRB.to.St.Jacques.LFRN.zip)

I make a flight from LFRB07R to LFRN010, flight level 60

### Expected behavior

The current leg before making the turn should not changed direction as I can see (it may follow the turn, but not so much and the angle between the 2 legs should remain the same)

### Steps to reproduce

See above.

### References (optional)

None.

### Additional info (optional)

I am not sure I have seen this with other waypoints.

### Discord Username (optional)

_No response_

### LINKED ISSUE #5042: Direct to a WPT erases whole flight plan in MCDU
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-06-06T15:12:39+00:00",
  "ref": "master",
  "sha": "ff9425ae19d607662e1e8adb6732dd0509aacb9f",
  "actor": "aguther",
  "event_name": "manual"
}
```

### Describe the bug

Requested direct to a waypoint and flight plan got erased in MCDU.

### Expected behavior

Direct to a waypoint should erase waypoints up to the direct and not the whole flight plan

### Steps to reproduce

1. Load a flightplan and take off.
2. Request a direct to enroute waypoint. Flight plan gets erased.

### References (optional)

image
image

### Additional info (optional)

This was working better before since it at least didn't erase the whole flight plan. I don't know how much is this related to MSDS own issues.

### Discord Username (optional)

_No response_

### LINKED ISSUE #5088: 2.5nm Range ring does not appear when TCAS is ON!
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-06-10T09:49:19+00:00",
  "ref": "master",
  "sha": "1578f4fa852800ef1d7c46ad6460f21bf92a2c3a",
  "actor": "MisterChocker",
  "event_name": "manual"
}
```

### Describe the bug

Turning ON the TCAS doesn't show the 2.5nm Range ring.

### Expected behavior

When TCAS is turned ON (TA or TA/RA), a 2.5nm distinct range ring should come up.

### Steps to reproduce

1. Position the aircraft on the runway
2. Select range to 10/20nm
3. Turn TCAS to TA or TA/RA
4. A 2.5nm range ring is expected if 20nm selected. 
5. The 2.5nm range ring is expected to change from continuosly dashed lines to vertical widely separated lines if 10nm is selected.

### References (optional)

TCAS
TCAS2

### Additional info (optional)

_No response_

### Discord Username (optional)

Illuminati

### LINKED ISSUE #5121: Performance affected by inserting a flight plan
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-06-14T09:02:20+00:00",
  "ref": "master",
  "sha": "4eaf1e43033d8ce8ae96027ab4449462121417fc",
  "actor": "MisterChocker",
  "event_name": "manual"
}
```

### Describe the bug

Significant drop in performance after inserting a flight plan into the MCDU (both from Simbrief and manual input). FPS remains unaffected, but the overall smoothness of rendering is affected. The issue is most visible in outside view while circling the view around the plane. The camera movement is smooth at the beginning, but at certain point the rendering freezes and then rapidly resumes.
The performance parameters as CPU and GPU utilization or FPS count are not affected after inserting a flight plan (when this behaviour beggins).

The only way to reduce this performance hit is by turning off the ND displays, but it helps only a little. Once the plane is powered down, the issue is no longer present.

The longer the flight plan, the bigger the performance drop. The performance drop is present regardless of the used graphics settings (tested also on absolutely minumum settings without any effect on the issue).

It looks like the flightplan being added affects the coherentGT draw the most. The issue affects both Stable and Developmental versions.

### Expected behavior

To be able to load a flight plan without a significant performance drop again.

### Steps to reproduce

1. Load the aircraft
2. Turn on the main systems (External power, ADIRS etc.)
3. Enter a flight plan in the MCDU and pan the camera around in both inside and outside views
4. Try turning off the ND displays and observe if it helps (in my case it sometimes does)
5. Power down the aircraft and notice the performance stabilise
6. EXTRA STEP: I have noticed a small performance drop also after turning on the flyPad

### References (optional)

4 videos demonstrating the issue uploaded to Google Drive:

https://drive.google.com/drive/folders/1PHUlQCVyCTpApepb1szP1rPYAxgtGeKH?usp=sharing

1. File: MSFS FBW PERFORMANCE_SHORTER FP - Demonstration of the issue using a shorter route and default livery
2. File: MSFS FBW PERFORMANCE_VERY LONG FP - Demonstration of the issue using a long route and default livery
3. File: MSFS FBW PERFORMANCE_DEPARTURE_SHORTER FP - Demonstration of the issue using a shorter route during taxi and departure in a custom livery
4. File: MSFS FBW PERFORMANCE_CRUISE_SHORTER FP - Demonstration of the issue using a shorter route during cruise in custom livery

The additional custom livery videos are provided to prove that a custom livery has no impact on the issue. Tested also with an empty Community folder with the same results.

### Additional info (optional)

Example of a long route with the biggest performance hit (used in the two on ground videos from References):

LZIB/RWY13 TABI1Y TABIN Q350 ERGOM DCT BUD DCT OSLEN DCT TEGRI L605 BULEN L744 UDROS M859 KARDE N644 LAGAS A910 BARAD M747 SUBUT T923 ERLEV M11 RODAR A909 BABUM A477 NONIK B361 DUTAR A845 DAVET P173 TAPIS L509 LAJAK G796 HANGU J139 NONIB L509 SULOM A466 ASARI J24 SAM J21 DPN R460 LKN G598 SMR SMRA2R RNAV02/RATAN/VNKT

Example of a shorter/mid range route with lower performance hit (used in the 2 in flight videos from References):

LZIB/RWY13 BERV1A BERVA P10 UPLAV L156 SOPAV L617 ADVAB L619 DENKO L996 VAVUN M602 BINKA SONAL KOVIK GUNPA 63N010W ING G3 NASBU NASB4N/ILS10Z/SONAX/BIKF

Hardware used:
CPU - AMD Ryzen 5 5600X
GPU - Palit Gamerock NVIDIA RTX 3070
RAM - Patriot Viper Steel Series 32 GB Dual Channel 3200 MHz
SIM DRIVE - Samsung EVO 970 1 TB M.2 NVME
SYSTEM DRIVE - ADATA XPG 8200 PRO 512 GB M.2 NVME
MONITOR - Native 1440p G-SYNC Compatible

### Discord Username (optional)

Liliputan#8440

### PR DESCRIPTION
**Pull request by Benjozork**
First part of #4028

Fixes #939
Fixes #1388
Fixes #2848
Fixes #4175
Fixes #1385
Fixes #5037
Fixes #5042
Fixes #5121
Fixes #4639
Fixes #4682
Fixes #2848 
Fixes #2683
Fixes #5088 
Fixes #2893 
Fixes #2350 
Fixes #2345 

## Summary of Changes

* [x] Implement custom flight plan manager based on WT implementation
* [x] Adapt fix naming to Airbus standards
* [x] Discontinuities
  * [x] Basic implementation
  * [x] Effect on NAV mode
* [x] DIRECT TO
  * [x] Basic implementation
  * [x] Sequencing 
* [x] LNAV
  * [x] Implement flight path geometry (credit to @beheh)
  * [x] Implement guidance laws (credit to @beheh)
* [ ] VNAV
  * [ ] Restore base functionality (DECEL) 
* [ ] F-PLN page
  * [x] Support discontinuities
    * [x] On regular waypoints
    * [ ] On origin 
  * [x] Support editing / removal of waypoints in STARs
  * [x] Fix all non-destination waypoints disappearing on a DIRECT TO to the last waypoint before the destination
  * [x] Fix inability to select STAR without transition or via
* [ ] Redo ND
* [x] Fix airway entry

Discord username (if different from GitHub): someperson#4953

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Benjozork** [comment]
I expect this to be ready for review in a few weeks

**Comment by Benjozork** [comment]
@solarisfire this is a draft PR, it is not integrated into the plane. It physically cannot cause your issues

File: src/shared/src/MathUtils.ts [hunk 1]
```diff
@@ -18,4 +18,15 @@ export class MathUtils {
 
        return (Math.round(val * coefficient) / coefficient).toString();
    }
+
+   public static diffAngle(a: number, b: number): number {
```
Comment (tracernz) [inline src/shared/src/MathUtils.ts:22]: /**
  * @returns difference between angle b and angle a, normalised to (-180, 180]
  */

**Comment by Benjozork** [comment]
@Fernien What does that have to do with this PR ?

**Comment by Fernien** [comment]
> @Fernien What does that have to do with this PR ?

I had a couple of tabs open. Apparently i posted it in the wrong one. My bad.

**Comment by tracernz** [comment]
Experimental isn't going to be used, so closing this one.

**Pull request closed without merging**